code/game/machinery/bank_machine.dm

```diff
@@ -67,6 +67,10 @@
 		say("Insufficient power. Halting siphon.")
 		end_siphon()
 		return
+	if (!(is_station_level(src.z) || is_centcom_level(src.z)))
+		say("Error: Console not in reach of station. Siphon halted.")
+		end_siphon()
+		return
```

Comment:
This is not quite what I said
You should still prevent people from enabling siphoning while off Z, you just need to implement `on_changed_z_level` -> `if not valid z, end_siphon`